uftrace.c

```diff
@@ -1474,7 +1474,16 @@ int main(int argc, char *argv[])
 		debug = 1;
 
 	if (opts.logfile) {
-		logfp = fopen(opts.logfile, "a");
+		char *logfile_path = NULL;
+		if (create_directory(opts.dirname) < 0) {
```

Comment:
I'm not sure about this.  Creating directory includes renaming the existing directory.  This can be a problem either you create a new one for record or dealing with an existing one for replay.